EasyPost.Tests.Net/EventTest.cs

```diff
@@ -7,29 +8,67 @@ namespace EasyPost.Tests.Net
     public class EventTest
     {
         [TestInitialize]
-        public void Initialize() => ClientManager.SetCurrent("NvBX2hFF44SVvTPtYjF0zQ");
+        public void Initialize()
+        {
+            VCR.SetUp(VCRApiKey.Test, "event", true);
+        }
+
+        private static EventCollection GetBasicEventCollection()
+        {
+            return Event.All(new Dictionary<string, object>
+            {
+                {
+                    "page_size", Fixture.PageSize
+                }
+            });
+        }
 
         [TestMethod]
-        public void TestLoad() => Assert.AreEqual(Resource.Load<Event>("{'id': 'barfoo'}").id, "barfoo");
+        public void TestAll()
+        {
+            VCR.Replay("all");
+
+            EventCollection eventCollection = GetBasicEventCollection();
+
+            List<Event> events = eventCollection.events;
+
+            Assert.IsTrue(events.Count <= Fixture.PageSize);
+            Assert.IsNotNull(eventCollection.has_more);
+            foreach (var item in events)
+            {
+                Assert.IsInstanceOfType(item, typeof(Event));
+            }
+        }
 
         [TestMethod]
-        [ExpectedException(typeof(HttpException))]
         public void TestRetrieve()
         {
-            // Events are archived after some time. Lets at least make sure we get a 404.
-            Event e = Event.Retrieve("evt_d0000c1a9c6c4614949af6931ea9fac8");
+            VCR.Replay("retrieve");
+
+            // use "TestAll"
+            EventCollection eventCollection = GetBasicEventCollection();
+            Event _event = eventCollection.events[0];
+
+            Event retrievedEvent = Event.Retrieve(_event.id);
+
+            Assert.IsInstanceOfType(retrievedEvent, typeof(Event));
+            Assert.AreEqual(_event.id, retrievedEvent.id);
         }
 
         [TestMethod]
-        public void TestRetrieveAll()
+        public void TestRetrieveBadInput()
         {
-            EventCollection eventCollection = Event.All();
-            Assert.IsNotNull(eventCollection);
-            foreach (Event tEvent in eventCollection.events)
-            {
-                Assert.IsNotNull(tEvent.id);
-                Assert.AreEqual(tEvent.id.Substring(0, 4), "evt_");
-            }
+            VCR.Replay("retrieve_bad_input");
```

Comment:
This shouldn't need a VCR since it raises an exception locally prior to making an HTTP request. Can we remove the associated cassette if it exists?

Also, these tests were specific to the PHP lib, if C# doesn't have this kind of logic then no need to test it.